xapian-core/include/xapian/diversify.h

```diff
@@ -0,0 +1,124 @@
+/** @file diversify.h
+ *  @brief Diversification API
+ */
+/* Copyright (C) 2018 Uppinder Chugh
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#ifndef XAPIAN_INCLUDED_DIVERSIFY_H
+#define XAPIAN_INCLUDED_DIVERSIFY_H
+
+#if !defined XAPIAN_IN_XAPIAN_H && !defined XAPIAN_LIB_BUILD
+#error "Never use <xapian/diversify.h> directly; include <xapian.h> instead."
+#endif
+
+#include "xapian/cluster.h"
+#include <xapian/attributes.h>
+#include <xapian/mset.h>
+#include <xapian/types.h>
+#include <xapian/visibility.h>
```

Comment:
Should be `<>` for all of these in an installed header, and they should be in sorted order.

Reply:
Changes pushed in new commit.